### ISSUE TITLE
Request: Delete action for puush

### ISSUE THREAD

**Issue by Ihysoal** [ShareX/ShareX#2215]
I stepped over from the puush client but I'm missing the fast way to delete, maybe you could add it at some point?

On that note, I'm dying with curiosity how you would do it, I've been browsing the code for a good few hours seeing if I could do it. I've given up.

**Comment by Jaex** [comment]
Are you sure puush API supports deletion?

**Comment by Ihysoal** [comment]
Aha! For you see, I did my research and stumbled upon [this](https://github.com/blha303/puush-linux/blob/master/apiDocumentation.md).

... Also, the puush client itself supports deletion.

**Comment by Jaex** [comment]
Unfortunately using "Deletion URL" support of ShareX not possible because that delete API request not supports get web requests (which happens when you open url in browser).

**Comment by Ihysoal** [comment]
How do other uploaders do it, e.g. dropbox?

**Comment by Jaex** [comment]
Dropbox don't have deletion url.

**Comment by diksm8** [comment]
Uploading to puush using the API uses a POST request, the same type of request that is used to delete an image.

**Comment by Jaex** [comment]
I added DeleteFile function to https://github.com/ShareX/ShareX/blob/master/ShareX.UploadersLib/FileUploaders/Puush.cs yesterday so I know it requires POST request. But there is many difficulties to implement that in multi uploader supported software.

**Comment by Ihysoal** [comment]
That's what I found out too, so I went on a search in the sources to see what the other file uploaders do because they all have delete functions too. I never tried to actually use them in ShareX. Turns out they aren't actually available.

I couldn't really find an entry point where you'd implement such a thing so I ended up asking.

Thanks for giving it a go at least.

**Comment by Scrxtchy** [comment]
I'm going to raise my hand and ask for Deletion over HTTP verbs be added, as a number of uploads would benefit from this

**Comment by Jaex** [comment]
I also want SFTP file deletion support!

**Comment by Scrxtchy** [comment]
Deletion for sftp is pretty simple, however we still would require something to handle making these requests.
for sftp though, you'll need to remember is entire path, basic `rm` would handle the rest.

Haven't used anything to do with ftp in sharex for years though, moved to http

**Comment by Jaex** [comment]
You are wrong just remembering path is not enough. You also gonna need to remember entire FTP account info. In case that user change FTP settings, it must not remove path from another FTP account. So a lot of configuration must be remembered to not incorrectly remove files.

**Comment by Scrxtchy** [comment]
If people are changing settings to an ftp identity they should expect that this would change the data of which their previous uploads are stored. As goes for if they move them from the shell directly.

There's a few different ways that this could be tackled really
would help if there was an sftp browser as a fallback option for when `Couldn't delete file: No such file or directory` is returned
